qt/scientific_interfaces/General/StepScan.cpp

```diff
@@ -695,6 +717,26 @@ void StepScan::plotCurve() {
                        yAxisTitle + "')";
 
   runPythonCode(QString::fromStdString(pyCode));
+#else
+  using namespace MantidQt::Widgets::MplCpp;
+  using namespace MantidQt::Widgets::Common;
+  auto ws = AnalysisDataService::Instance().retrieveWS<Mantid::API::MatrixWorkspace>(m_plotWSName);
+  Mantid::PythonInterface::GlobalInterpreterLock lock;
+  Python::Object module{Python::NewRef(PyImport_ImportModule("mantid.plots"))};
+  auto canvas = new FigureCanvasQt(111, "mantid");
+  auto fig = canvas->gcf();
+  auto ax = canvas->gca<MantidAxes>();
+  title += " - Step Scan";
+  canvas->setWindowTitle(QString::fromStdString(title));
+  QHash<QString, QVariant> hash;
+  hash.insert("linestyle", "");
+  hash.insert("marker", ".");
+  auto line = ax.plot(ws, 0, "black", "", hash);
+  ax.setXLabel(xAxisTitle.c_str());
+  ax.setYLabel(yAxisTitle.c_str());
+  canvas->draw();
+  canvas->show();
```

Comment:
@martyngigg Is there a straightforward way to add the Workbench's plot toolbar and connect the figure to the plots toolbox?

Reply:
This is actually one thing that is not as easy yet. The toolbar is defined in [workbench.plotting](https://github.com/mantidproject/mantid/blob/ed972e979ee49c0996485dc52120c4f153309b0e/qt/applications/workbench/workbench/plotting/toolbar.py) so that module should probably be moved to [mantidqt.plotting](https://github.com/mantidproject/mantid/tree/ed972e979ee49c0996485dc52120c4f153309b0e/qt/python/mantidqt/plotting) to it can be reused by other things that can be run standalone outside of workbench.
You should be then able to import that class and use it with your canvas.
Let me know how this goes and if I can help at all with it.